Sources/MapboxDirectionsCLI/CodingOperation.swift

```diff
@@ -57,8 +129,17 @@ class CodingOperation<ResponceType : Codable, OptionsType : DirectionsOptions >
         
         decoder.userInfo = [.options: directionsOptions,
                             .credentials: BogusCredentials]
+        
+        var directionsResultsProvider: DirectionsResultsProvider?
+        if options.outputFormat == .gpx {
+            do {
+                directionsResultsProvider = try decoder.decode(ResponceType.self, from: input) as DirectionsResultsProvider
+            } catch {
+                directionsResultsProvider = try decoder.decode(MapMatchingResponse.self, from: input) as DirectionsResultsProvider
```

Comment:
Do we really need to fallback to `MapMatchingResponse` here? The command will fail in `processResponse` in case data is not decodable with `ResponceType`.